ports/samd/machine_spi.c

```diff
@@ -94,9 +102,15 @@ static void machine_spi_init(mp_obj_base_t *self_in, size_t n_args, const mp_obj
         { MP_QSTR_polarity, MP_ARG_INT, {.u_int = -1} },
         { MP_QSTR_phase, MP_ARG_INT, {.u_int = -1} },
         { MP_QSTR_firstbit, MP_ARG_INT, {.u_int = -1} },
+        #if defined(pin_SCK) && defined(pin_MOSI) && defined(pin_MISO)
+        { MP_QSTR_sck, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = pin_SCK} },
+        { MP_QSTR_mosi, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = pin_MOSI} },
+        { MP_QSTR_miso, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = pin_MISO} },
+        #else
         { MP_QSTR_sck, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
         { MP_QSTR_mosi, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
-        { MP_QSTR_miso, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_rom_obj = MP_ROM_NONE} },
+        { MP_QSTR_miso, MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_int = -1} },
```

Comment:
This use of `MP_ARG_OBJ` with `.u_int` is invalid.  Needs to be one of:
1. `MP_ARG_INT` and `.u_int = -1`
2. `MP_ARG_OBJ` and `.u_rom_obj = MP_ROM_INT(-1)`

I think you want the latter, so the user can pass in a pin object.

Reply:
OK. Somehow it worked before.

Reply:
Looking at it, the -1 case can be dropped. Setting the default to MP_CONST_NONE is sufficient for the logic. And it simplifies the code.